### PR TITLE
Add test for PeerIdTooLong and clean up code

### PR DESCRIPTION
**Pull request by HasanZaigam**
### **PR Title:**
Add Test for PeerIdTooLong Error and Clean Up Code in Network Authorization Pallet

---

### **Description:**

This PR enhances the `pallet-network-authorization` module by adding tests for the `PeerIdTooLong` error scenario and refactoring the code to improve readability and maintainability.

#### **Approach:**
1. **Test for PeerIdTooLong Error:** A test case is added to ensure that when a peer ID exceeds 64 bytes, it triggers the `PeerIdTooLong` error, validating the correct error handling.
2. **Successful Authorization Test:** Another test ensures that a valid peer ID (64 bytes) is correctly authorized, confirming the function works as expected for valid inputs.
3. **Code Cleanup:** Removed commented-out code to improve clarity and maintainability. The mock environment setup was also simplified.

#### **Why This Matters:**
- **Better Test Coverage:** This PR ensures edge cases are covered, specifically for error handling related to peer ID validation.
- **Code Quality:** Cleaning up commented code improves readability, making the codebase more maintainable.
- **Consistency:** These changes ensure the pallet behaves consistently when handling valid and invalid peer IDs.

---

### **How to Test:**
To run the tests, use the following command:
```bash
cargo test -p pallet-network-authorization -- --nocapture
```

This will verify that the error handling and successful authorization logic are working correctly.

---

By merging this PR, the code becomes more robust and easier to maintain, while laying the foundation for additional tests in the future.

### CODE REVIEW

**Comment by vatsa287** [comment]
@HasanZaigam 
The test `PeerIdTooLong` belongs to `node-authorisation` module.
There is already a stale PR on this, refer it and re-comit it.
https://github.com/dhiway/cord/pull/499

**Comment by HasanZaigam** [comment]
Hi @vatsa287, I am confused now about what I have to do. Please tell me, if possible, if you can take a Google Meet. I didn't understand what I had to do next.

**Review by Hasan2k5 — COMMENTED** [review]
Looks good overall! The function is clear, and the length check for `peer_id` is well-implemented. Just noticed some minor indentation issues—fixing that would make it more readable. Nice work!

**Comment by vatsa287** [comment]
@HasanZaigam @Hasan2k5 
You can ping me on discord `vatsa_dhiway`, for more clarifications.

File: Cargo.lock [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # This file is automatically @generated by Cargo.
```
Comment (vatsa287) [inline Cargo.lock:1]: This file is not required.

File: Cargo.toml [hunk 2]
```diff
@@ -20,6 +20,7 @@ members = [
 	"pallets/did",
```
Comment (vatsa287) [inline Cargo.toml]: Changes made here is not required as well.

File: test-utils/service/src/lib.rs [hunk 3]
```diff
@@ -178,7 +178,7 @@ where
 				interval.tick().await;
 
 				if full_nodes.iter().all(|(id, service, _, _)| full_predicate(*id, service)) {
-					break
+					break;
```
Comment (vatsa287) [inline test-utils/service/src/lib.rs:181]: This change is not required as it is the last statement in the block in Rust.

File: pallets/node-authorization/src/tests.rs [hunk 4]
```diff
@@ -459,3 +459,19 @@ fn test_generate_peer_id_invalid_utf8() {
 	let invalid_node_id: NodeId = vec![0xFF, 0xFE, 0xFD];
 	assert_err!(NodeAuthorization::generate_peer_id(&invalid_node_id), Error::<Test>::InvalidUtf8);
 }
+
+// the test to check that the PeerId not too long.
+#[test]
```
Comment (vatsa287) [inline pallets/node-authorization/src/tests.rs:472]: This needs a rework.
Test all methods in `node-authorization` which emits `PeerIdTooLong`.

**Review by vatsa287 — CHANGES_REQUESTED** [review]

**Review by HasanZaigam — COMMENTED** [review]
I have made all the changes and improved the test.rs code to make the test for PeerIdTooLong more comprehensive, as per your guidance. Please review it again, sir.

**Comment by HasanZaigam** [comment]
@vatsa287, please check the changes and let me know , what next changes I have to make in it, please guide me .

**Comment by vatsa287** [comment]
@HasanZaigam 
All other changes other than `tests.rs` is not required.

**Comment by vatsa287** [comment]
> @HasanZaigam All other changes other than `tests.rs` is not required.

@HasanZaigam ^^

**Comment by HasanZaigam** [comment]
ok , i makes all changes write and add the test case in test only and let you know, after making the changes.

**Comment by HasanZaigam** [comment]
@vatsa287 makes the changes please review it
